posthog/temporal/data_imports/pipelines/pipeline/pipeline.py

```diff
@@ -250,6 +259,32 @@ def _process_pa_table(self, pa_table: pa.Table, index: int):
 
         _update_job_row_count(self._job.id, pa_table.num_rows, self._logger)
 
+        # if first sync, we make the delta table files available for querying and create the data warehouse table, so
+        # that the user has some data available to start using
+        if is_first_sync:
+            self._logger.debug(
+                "First sync, making delta table files available for querying and creating data warehouse table"
+            )
+            file_uris = delta_table.file_uris()
+            new_file_uris = [uri for uri in file_uris if uri not in previous_file_uris]
+            self._logger.debug(f"Adding {len(new_file_uris)} S3 files to query folder")
+            prepare_s3_files_for_querying(
+                folder_path=self._job.folder_path(),
+                table_name=self._resource_name,
+                file_uris=new_file_uris,
+                # delete existing files if it's the first chunk, otherwise we'll just append to the existing files
+                delete_existing=False if index > 0 else True,
+            )
+            self._logger.debug("Validating schema and updating table")
+            validate_schema_and_update_table_sync(
+                run_id=str(self._job.id),
+                team_id=self._job.team_id,
+                schema_id=self._schema.id,
+                table_schema_dict=self._internal_schema.to_hogql_types(),
+                row_count=row_count,
+                table_format=DataWarehouseTable.TableFormat.DeltaS3Wrapper,
+            )
```

Comment:
As a whole I think this logic is sound. But it heavily relies on us not changing how we're writing to the delta table. On the first sync, we always `append` to the table instead of `merging` into it, meaning on every iteration of the pipeline we're only adding new data files. if we were `merging` instead, then we'd be updating/rewriting/deleting data files in the delta table which won't be reflected in the querying folder, and thus we'd get duplicate data in there. 

The point being is this logic will fail if we change logic elsewhere unrelated to this, which I'm a lil worried about, especially given how hard it is to unit test this. Not sure what to do here other than maybe add a comment on the appending code or something - this just feels a bit brittle right now.

Reply:
yeah, I agree. In the case of a merge do you know if the file uris would change?

At the moment I am just doing:
```python
new_file_uris = [uri for uri in file_uris if uri not in previous_file_uris]
```

but I could also check if any `previous_file_uris` are not in `new_file_uris` and raise an error if so. I don't know enough about this to know if that would work but I guess I could test it locally. Otherwise I'll just have to add a comment like you say. Or I could create a new method, separate from `prepare_s3_files_for_querying` to make it more explicit?

Reply:
> In the case of a merge do you know if the file uris would change?

Yep, the files get rewritten, so the old file will be set as "to be deleted" (which table vaccumming takes care of), and then the new file is what will be used. So you'd find yourself with both the old and the new file in the query folder resulting in duplicate data 

> but I could also check if any previous_file_uris are not in new_file_uris and raise an error if so

yeah this would be good - maybe not error, but just skip the copying across and capture the error so we're aware that the feature is no longer enabled or something